opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
Please double check this, and the next few changes.  When I originally wrote this code, I remember thinking that I needed to make explicit copies, and not references. for some reason or another.  I agonized over this, because making copies hurts performance.   So ... maybe your changes are correct, but that is not obviously the case just by glancing at this.

Re: my earlier remarks about how unit tests test everything? Its possible that the unit tests do not actually exercise this code enough.